### PR TITLE
Changed word embedding technique from Tf-Idf to Word2Vec

### LINKED ISSUE #48: Modification of word embedding technique from Tf-Idf to Word2Vec
### Prerequisites

- [X] I am an Open Source Contributor
- [X] I checked to make sure that this issue has not already been filed

### Description

I want to apply word2vec embedding technique and observe the corresponding change in accuracy. Along with it, I want to apply GridSearchCV to see if it improves accuracy of the already applied algorithm

### Screenshots

_No response_

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/Rakesh9100/ML-Project-Drug-Review-Dataset/blob/main/.github/CONTRIBUTING_GUIDELINES.md) of this project.

### PR DESCRIPTION
**Pull request by Juhibhojani**
# Fixes Issue🛠️

Closes #48 

# Description👨‍💻 

# Type of change📄
Change of Word Embedding Technique (TfIdf to Word2Vec Vectorizer)
Steps performed:
1) Creating two new data frames (df for train data and df1 for test data) for performing different actions
2) Applying tokenization
3) Applying Stemming (Both of them use the Gensim library)
4) Training Word2Vec on given vocabulary
5) Converting into the vector using trained Word2Vec model (Here, for each word, a 1000*1 dimensional vector is created, and for each review i.e. each sentence, the average of all word's vectors is taken)
6) These vector is stored in a new file created by the code which is then read using read_csv function
7) After these, the review vector is merged with the rest of the columns, and unnecessary columns are dropped.
8) The application of model part is same 

- [] New feature (non-breaking change which adds functionality)

# How this has been tested✅

 I ran the entire code and it's error-free

# Checklist✅ 

- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my own code
- [] I have commented on my code, particularly in hard-to-understand areas
- [] I have added a demonstration in the form of a GIF/video file (No)
- [] I am an Open Source Contributor

# Screenshots/GIF📷

### CODE REVIEW

**Comment by Rakesh9100** [comment]
## Encoding the categorical columns
for i in ["drugName", "condition"]:
    df[i] = LabelEncoder().fit_transform(df[i])
    df1[i] = LabelEncoder().fit_transform(df1[i])
    
If you change the variable here, then every applied model will be affected.

Pls provide the screenshots of everya accuracy and plots output after your changes @Juhibhojani

**Comment by Juhibhojani** [comment]
Okay, I will but this variable change won't affect any other model, as each model's input is X_train and Y_train which I have made using df and df1 respectively.
I will be adding Screenshots soon as I have to run the model again! 

There is a slight increase in the accuracy of the model as compared to the previous Decision Tree classifier and a decrease in accuracy with linear regression. For other models its nearly the same!

**Comment by Juhibhojani** [comment]
DTC-1
DTC-2
DTC-3
DTC-4
Logistic-1
Logistic-2
logistic-3
LR-1
LR-2
LR-3
LR-4
LR-5
MLP-1
MLP-2
MLP-3
MLP-4

**Comment by Neilblaze** [comment]
@Juhibhojani can you please resolve the conflicts? Thanks! :)

**Comment by Juhibhojani** [comment]
I made changes in the requirements.txt file and I think this will resolve the conflict here! Let me know if there are any other changes!

**Review by Juhibhojani — COMMENTED** [review]
I have added Gensim in the requirements.txt file!

**Comment by Neilblaze** [comment]
@Juhibhojani It's still facing merge conflicts. Also, it's recommended to always checkout a new branch and push the changes into that, & **not** touching `main`. You might receive merge conflicts and linting issues if your PR is merging changes from `main` branch of your forked repo to the `main` branch of the parent repo.

**Comment by Juhibhojani** [comment]
I have made the necessary changes.

**Comment by Neilblaze** [comment]
@Rakesh9100 any idea why the lint is failing? 🤔

**Comment by Rakesh9100** [comment]
Suggesting to do `black main.py` on your local and it will format the code according to the black lint and then check will be passed @Neilblaze

**Comment by Rakesh9100** [comment]
It's not good to understand if we keep names df and df1. As it was previously train_imp and test_imp which is properly understandable so kindly update.

And after your changes run `black main.py` to format the code and then commit. @Juhibhojani

**Comment by Juhibhojani** [comment]
Okay.

**Comment by Rakesh9100** [comment]
Lint check is failing. To resolve pls do `black main.py` on your local to format the code and then commit.

And also do provide the output and confirmation that program is running flawlessly after your changes?

**Comment by Juhibhojani** [comment]
image
Ran black main.py and post changes the file is working properly
image

**Comment by Rakesh9100** [comment]
Pls change the PR title similar to the work you have done @Juhibhojani

**Comment by Rakesh9100** [comment]
Resolve the conflict in `requirements.txt` file pls @Juhibhojani

**Comment by Juhibhojani** [comment]
Done

**Comment by Rakesh9100** [comment]
LGTM!! Nice work. Congrats on getting your PR merged to this repo under GSSoC23 🥳🎉 @Juhibhojani

**Comment by Juhibhojani** [comment]
Thanks for all the help, It was really good working on this project!🙌
